### ISSUE TITLE
Segfault after starting DrRacket

### ISSUE THREAD

**Issue by apreiml** [swaywm/sway#2475]
Again after starting DrRacket. I'm currently at 98ed05225e06618533fa0439fdc83463cb9f1d2e 

```
#0  0x0000558af9f16b34 in output_surface_for_each_surface (output=0x558afb48ab10, surface=0x0, ox=1282, oy=48, iterator=0x558af9f18405 <render_surface_iterator>, user_data=0x7ffd719300b0)
    at ../../sway/desktop/output.c:131
#1  0x0000558af9f18965 in render_view_toplevels (view=
    0x558afb7bb6c0, output=0x558afb48ab10, damage=0x7ffd719302f0, alpha=1)
    at ../../sway/desktop/render.c:196
#2  0x0000558af9f18ca9 in render_view (output=0x558afb48ab10, damage=0x7ffd719302f0, con=0x558afb498e30, colors=0x558afb330ee4) at ../../sway/desktop/render.c:266
#3  0x0000558af9f1a031 in render_container_simple (output=0x558afb48ab10, damage=0x7ffd719302f0, con=0x558afb48ca50, parent_focused=false) at ../../sway/desktop/render.c:597
#4  0x0000558af9f1a779 in render_container (output=0x558afb48ab10, damage=0x7ffd719302f0, con=0x558afb48ca50, parent_focused=false) at ../../sway/desktop/render.c:745
#5  0x0000558af9f1aef0 in output_render (output=0x558afb48ab10, when=0x7ffd719302e0, damage=0x7ffd719302f0) at ../../sway/desktop/render.c:896
#6  0x0000558af9f1761a in damage_handle_frame (listener=0x558afb48ac08, data=0x558afb48d6b0)
    at ../../sway/desktop/output.c:392
#7  0x00007fea1be9263e in wlr_signal_emit_safe (signal=0x558afb48d710, data=0x558afb48d6b0)
    at ../../util/signal.c:29
#8  0x00007fea1be7ebd7 in output_handle_frame (listener=0x558afb48d7a8, data=0x558afb485b60)
    at ../../types/wlr_output_damage.c:50
#9  0x00007fea1be9263e in wlr_signal_emit_safe (signal=0x558afb485c78, data=0x558afb485b60)
    at ../../util/signal.c:29
#10 0x00007fea1be82075 in wlr_output_send_frame (output=0x558afb485b60) at ../../types/wlr_output.c:533
#11 0x00007fea1be4f09e in page_flip_handler (fd=9, seq=316403, tv_sec=6897, tv_usec=91553, user=0x558afb485b60) at ../../backend/drm/drm.c:1031
#12 0x00007fea17ff84d5 in drmHandleEvent () at /usr/lib/libdrm.so.2
#13 0x00007fea1be4f10f in handle_drm_event (fd=9, mask=1, data=0x0) at ../../backend/drm/drm.c:1041
#14 0x00007fea1becca42 in wl_event_loop_dispatch () at /usr/lib/libwayland-server.so.0
#15 0x00007fea1becb14c in wl_display_run () at /usr/lib/libwayland-server.so.0
#16 0x0000558af9f14a7f in server_run (server=0x558af9f66a00 <server>) at ../../sway/server.c:163
#17 0x0000558af9f14202 in main (argc=1, argv=0x7ffd71930cb8) at ../../sway/main.c:439
```

**Comment by RyanDwyer** [comment]
I opened it and it hasn't crashed. Is there something I have to do to make it crash, and is it 100% reproducible?

**Comment by apreiml** [comment]
I did the following

1. open recent file (https://gist.github.com/apreiml/f62c257b2e90a49712d189d13dcfade3)
2. Move mouse around inside the editor window. 

There are overlay windows appearing and at some point it crashes, I'll try to figure out when exactly.

**Comment by apreiml** [comment]
Log output: 
```
Gdk-Message: 13:34:42.114: Window 0x55ad364a31a0 is a temporary window without parent, application will not be able to position it on screen.
Gdk-Message: 13:34:42.114: Window 0x55ad364a31a0 is a temporary window without parent, application will not be able to position it on screen.
2018-08-17 13:34:42 - [types/xdg_shell/wlr_xdg_surface.c:445] new xdg_surface 0x5564f82249f0 (res 0x5564f822ab50)
2018-08-17 13:34:42 - [sway/desktop/xdg_shell.c:452] New xdg_shell toplevel title='' app_id='drracket'
2018-08-17 13:34:42 - [sway/tree/root.c:169] Looking up workspace for pid 17681
2018-08-17 13:34:42 - [sway/tree/container.c:451] Adding new view 0x5564f8224b70: to container 0x5564f821e9f0 4 Chapter 2.scm - DrRacket
2018-08-17 13:34:42 - [sway/ipc-server.c:324] Sending window::new event
2018-08-17 13:34:42 - [sway/ipc-server.c:324] Sending window::focus event
2018-08-17 13:34:42 - [sway/ipc-server.c:324] Sending window::title event
2018-08-17 13:34:42 - [sway/tree/arrange.c:211] Usable area for ws: 1920x1177@0,23
2018-08-17 13:34:42 - [sway/tree/arrange.c:304] Removing gaps: not gapped: 0x5564f7f19c10
2018-08-17 13:34:42 - [sway/tree/arrange.c:337] Adding gaps: 0x5564f7f19c10
2018-08-17 13:34:42 - [sway/tree/arrange.c:220] Arranging workspace '2' at 0.000000, 23.000000
2018-08-17 13:34:42 - [sway/tree/arrange.c:165] Arranging layout for 0x5564f7f19c10 2 1920.000000x1177.000000+0.000000,23.000000
2018-08-17 13:34:42 - [sway/tree/arrange.c:304] Removing gaps: not gapped: 0x5564f821e9f0
2018-08-17 13:34:42 - [sway/tree/arrange.c:304] Removing gaps: not gapped: 0x5564f8224b70
2018-08-17 13:34:42 - [sway/tree/arrange.c:48] Arranging 0x5564f7f19c10 horizontally
2018-08-17 13:34:42 - [sway/tree/arrange.c:54] Calculating arrangement for 0x5564f821e9f0:4 (will scale 1920.000000 by 0.500000)
2018-08-17 13:34:42 - [sway/tree/arrange.c:337] Adding gaps: 0x5564f821e9f0
2018-08-17 13:34:42 - [sway/tree/arrange.c:54] Calculating arrangement for 0x5564f8224b70:4 (will scale 1920.000000 by 0.500000)
2018-08-17 13:34:42 - [sway/tree/arrange.c:337] Adding gaps: 0x5564f8224b70
2018-08-17 13:34:42 - [sway/desktop/transaction.c:275] Transaction 0x5564f81e0110 committing with 4 instructions
2018-08-17 13:34:42 - [sway/ipc-server.c:324] Sending window::focus event
2018-08-17 13:34:42 - [sway/ipc-server.c:324] Sending window::close event
2018-08-17 13:34:42 - [sway/tree/arrange.c:211] Usable area for ws: 1920x1177@0,23
2018-08-17 13:34:42 - [sway/tree/arrange.c:304] Removing gaps: not gapped: 0x5564f7f19c10
2018-08-17 13:34:42 - [sway/tree/arrange.c:337] Adding gaps: 0x5564f7f19c10
2018-08-17 13:34:42 - [sway/tree/arrange.c:220] Arranging workspace '2' at 0.000000, 23.000000
2018-08-17 13:34:42 - [sway/tree/arrange.c:165] Arranging layout for 0x5564f7f19c10 2 1920.000000x1177.000000+0.000000,23.000000
2018-08-17 13:34:42 - [sway/tree/arrange.c:304] Removing gaps: not gapped: 0x5564f821e9f0
2018-08-17 13:34:42 - [sway/tree/arrange.c:48] Arranging 0x5564f7f19c10 horizontally
2018-08-17 13:34:42 - [sway/tree/arrange.c:54] Calculating arrangement for 0x5564f821e9f0:4 (will scale 960.000000 by 2.000000)
2018-08-17 13:34:42 - [sway/tree/arrange.c:337] Adding gaps: 0x5564f821e9f0
2018-08-17 13:34:42 - [sway/desktop/transaction.c:252] Transaction 0x5564f81e0110 timed out (1 waiting)
2018-08-17 13:34:42 - [sway/desktop/transaction.c:149] Applying transaction 0x5564f81e0110
2018-08-17 13:34:42 - [sway/desktop/transaction.c:275] Transaction 0x5564f8233010 committing with 4 instructions
failed to read Wayland events: Broken pipe
```

**Comment by RyanDwyer** [comment]
Where are these overlay windows? I loaded the file you supplied and moved the mouse around everywhere, and I'm noticing a significant lack of overlay windows. Maybe a plugin is responsible for them?

**Comment by apreiml** [comment]
With overlay windows I mean that there appear arrows when you move your mouse over keywords. The arrows point to the place where those where defined. Also a tooltip is flickering in the center of the screen. For me it crashes sometimes, but not reliably, when I move over keywords like `cons`.

I didn't install any addon beside the sicp language. I got drracket from the archlinux repo. It's included in community/racket 7.0-1

**Comment by RyanDwyer** [comment]
I opened it in rootston and now I see what you mean with the overlay windows. I didn't get that in sway at all. In fact there's so much stuff messed up when I run it in sway - not only do the overlay windows not appear, but sometimes the main area stops responding to cursor clicks. For example, I sometimes can't click to move the caret or click the Run button, but the File etc menus work fine. And trying to kill the program with a sway bindsym didn't work.

After running it in rootston, I then ran it in sway again and now I get the overlay windows. I noticed it's spamming the creation and deletion of a second window which I think is trying to float.

I'm going to leave this for now. I'll revisit it after my `seat_get_focus` PR is merged, and after I've removed the `L_FLOATING` container (no PR for this yet).
